doc/modules/computing.rst

```diff
@@ -504,20 +504,47 @@ Links
   - `Scipy sparse matrix formats documentation <https://docs.scipy.org/doc/scipy/reference/sparse.html>`_
 
 Parallelism, resource management, and configuration
-=====================================================
+===================================================
 
 .. _parallelism:
 
-Parallel and distributed computing
------------------------------------
+Parallelism
+-----------
 
-Scikit-learn uses the `joblib <https://joblib.readthedocs.io/en/latest/>`__
-library to enable parallel computing inside its estimators. See the
-joblib documentation for the switches to control parallel computing.
+Scikit-learn supports parallel implementations in two ways: via the `joblib
+<https://joblib.readthedocs.io/en/latest/>`_ library, or via OpenMP (through
+Cython, which allows to release the GIL).
+
+When the underlying implementation uses joblib, the number of jobs that are
+spawned in parallel can be controled via the ``n_jobs`` parameter. In most
+cases, scikit-learn uses joblib to rely on multi-processing rather than on
+multi-threading.
+
+OpenMP is used to parallelize low-level code written in Cython, relying on
+multi-threading exclusively. By default, the implementation will use as many
+threads as possible, while still avoiding over-subscription.
```

Comment:
I thought that was just fixed in the new joblib release?

Reply:
Only for process based parallelism. It can be done with the threading backend because it's based on environment variables which mkl, openblas, openmp,... read only once, when they're imported.

Reply:
For reference, here is the issue regarding using joblib threading and BLAS: https://github.com/scikit-learn/scikit-learn/issues/14979